⚙️BaseEntity 및 BaseTimeEntity 도입

sobee1403 opened the pull request:
📘 작업 내용
 - 공통 엔티티 추상 클래스 BaseEntity, BaseTimeEntity 추가
 - @MappedSuperclass 적용으로 상속받는 엔티티에서 자동으로 컬럼 매핑
 - 생성일(createdAt), 수정일(updatedAt) 등 공통 필드 및 추상화 구조 정의

🎯 도입 이유
 - 모든 엔티티에 공통으로 필요한 필드를 중앙 관리
 - 코드 중복 제거 및 일관된 엔티티 관리 구조 확립
 - Soft Delete 정책 등 확장성 고려 기반 마련
 
⚙️ 활용 방법
 - 각 도메인 엔티티가 BaseEntity 또는 BaseTimeEntity를 상속
 - 별도 컬럼 선언 없이 자동으로 생성·수정 시각 관리
 - 향후 삭제 정책(isDeleted, deletedAt) 등 공통 속성 확장 가능

dawwson commented on src/main/java/com/wanted/growthmate/common/entity/BaseTimeEntity.java:
```diff
@@ -0,0 +1,32 @@
+
+@MappedSuperclass
+@EntityListeners(AuditingEntityListener.class)
+public abstract class BaseTimeEntity {
```
[ 궁금한 점 🤔 ]
- 추상 메서드가 없다면 그냥 클래스로 선언해도 무방한 건가요?
- deleted_at도 추가하고 싶으면 필드만 추가하면 되나요!?

dawwson reviewed (approved):
고생하셨습니다. 이거 적용하면 공통 필드들을 다 뺄 수 있어서 편할 것 같아요!

sobee1403 committed:
core: SoftDeleteBaseEntity.java 추가

sobee1403 committed:
core: BaseEntity 구조 변경

sobee1403 committed:
refactor: BaseEntity 계층 구조 및 equals/hashCode 개선

- BaseEntity, BaseTimeEntity, SoftDeleteBaseEntity 상속 구조 분리
- Hibernate 프록시 안전 비교를 위한 equals()/hashCode() 수정
- createdAt/updatedAt 필드명 통일
- JPA 리플렉션 생성을 위한 기본 생성자(protected) 추가

sobee1403 committed:
Merge remote-tracking branch 'origin/develop' into feat/base-entity-timestamp
